### DISCUSSION TITLE
Issues with laptops sleeping and waking with system time

### DISCUSSION THREAD

**Discussion by karlmutch** [rancher-sandbox/rancher-desktop#836] [Q&A]
I have been using several services within k3s deployments including rancher-desktop and kim. Because I am running on a laptop the clock within various containers inside the cluster looses time while the machine is in a sleep state.

If I try to use ntpdate to synchronize things within containers I get issues with privileges and if I try to reset the clocks using hwclock also within containers I only ever see the delayed time that is affected by the host OS sleeping.

Some pods/containers I have benefit from having an ntpd deployment but the system pods etc never seem to recover the lost time.

How are we meant to be getting our clocks back into sync after sleeping etc ?

Question also asked at https://github.com/k3s-io/k3s/discussions/4256.  Given laptops etc are a prime use case for rancher desktop I'm not sure resetting Kubernetes is viable everytime the laptop sleeps.

**Comment by jandubois** [comment]
Is this on Windows or MacOS/Linux? It is probably a hypervisor thing to keep the clocks in sync, so I need to know which platform to look at.

**Comment by karlmutch** [comment]
Sorry OSX. rancher desktop 0.6.0.

**Comment by jandubois** [comment]
I've created #839 to track this; I believe it can/should be fixed at the supervisor level.

There may still be an issue that k3s may not like it if the time suddenly jumps forward by a large amount, but that is independent from the goal of keeping time in sync between the host and the guest.
